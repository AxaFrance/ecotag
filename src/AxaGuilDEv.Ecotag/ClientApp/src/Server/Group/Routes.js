import React from 'react';
import {Route, Switch} from 'react-router-dom';
import PageNotFound from '../NotFound';

const Routes = () => {
    return (
        <Switch>
            <Route component={PageNotFound}/>
        </Switch>
    );
};

export default Routes;
