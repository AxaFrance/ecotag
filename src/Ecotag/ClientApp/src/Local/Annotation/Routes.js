import React from "react";
import {Route, Switch, useRouteMatch, withRouter} from "react-router";
import AnnotationsContainer from "./AnnotationsContainer";
import TitleBar from "../../TitleBar";
import DatasetHandler from "./DatasetHandler";

const Annotation = props => {
    const {
        match: {
            params: {id, dataset},
        },
        annotationState,
        fetch,
        url
    } = props;
    return (
        <>
            <TitleBar
                title={dataset}
                goTo="/annotate"
                goTitle="Annotate"
            />
            <AnnotationsContainer
                state={annotationState}
                id={id}
                url={url}
                dataset={dataset}
                fetch={fetch}
            />
        </>);
}

const Annotate = ({fetch, state, setState}) => {
    return (
        <>
            <TitleBar title={`Annotate`}/>
            <DatasetHandler state={state} setState={setState} fetch={fetch}/>
        </>
    );
}

const AnnotationWithRouter = React.memo(withRouter(Annotation));

const Routes = ({state, setState, fetch}) => {
    const {url} = useRouteMatch();

    return (
        <Switch>
            <Route exact path={`${url}/:dataset/:id`}>
                <AnnotationWithRouter annotationState={state} url={url} fetch={fetch}/>
            </Route>
            <Route>
                <Annotate state={state} setState={setState} fetch={fetch}/>
            </Route>
        </Switch>
    );
};

export default React.memo(Routes);