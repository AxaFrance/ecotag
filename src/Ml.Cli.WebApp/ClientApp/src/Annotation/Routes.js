import React, {useEffect} from "react";
import {Route, Switch, useRouteMatch, withRouter} from "react-router-dom";
import AnnotationsContainer from "./AnnotationsContainer";


const Annotation = props => {
    const {
        match: {
            params: {id, dataset},
        },
        annotationState,
        fetchFunction,
        url
    } = props;
    return <AnnotationsContainer
        state={annotationState}
        id={id}
        url={url}
        dataset={dataset}
        fetchFunction={fetchFunction}
    />
}

const AnnotationWithRouter= React.memo(withRouter(Annotation));

const Routes = ({annotationState, fetchFunction}) => {
    const {url} = useRouteMatch();
    
    return(
        <Switch>
            <Route exact path={`${url}/:dataset/:id`} >
                <AnnotationWithRouter annotationState={annotationState} url={url} fetchFunction={fetchFunction}  />
            </Route>
        </Switch>
    );
};

export default React.memo(Routes);