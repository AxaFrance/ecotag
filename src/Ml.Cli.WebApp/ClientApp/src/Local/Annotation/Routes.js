import React from "react";
import {Route, Switch, useRouteMatch, withRouter} from "react-router";
import AnnotationsContainer from "./AnnotationsContainer";
import TitleBar from "../../TitleBar";
import DatasetHandler from "./DatasetHandler";

const Annotation = props => {
    const {
        match: {
            params: {id, dataset},
        },
        annotationState,
        fetchFunction,
        url
    } = props;
    return (
    <>
        <TitleBar
            title={dataset}
            goTo="/annotate"
            goTitle="Annotate"
        />
        <AnnotationsContainer
            state={annotationState}
            id={id}
            url={url}
            dataset={dataset}
            fetchFunction={fetchFunction}
        />
    </>);
}

const Annotate = ({fetchFunction, state, setState}) => {
    return (
        <>
            <TitleBar title={`Annotate`} />
            <DatasetHandler state={state} setState={setState} fetchFunction={fetchFunction}/>
        </>
    );
}

const AnnotationWithRouter= React.memo(withRouter(Annotation));

const Routes = ({state, setState, fetchFunction}) => {
    const {url} = useRouteMatch();
    
    return(
        <Switch>
            <Route exact path={`${url}/:dataset/:id`} >
                <AnnotationWithRouter annotationState={state} url={url} fetchFunction={fetchFunction}  />
            </Route>
            <Route>
                <Annotate state={state} setState={setState}  fetchFunction={fetchFunction}/>
            </Route>
        </Switch>
    );
};

export default React.memo(Routes);