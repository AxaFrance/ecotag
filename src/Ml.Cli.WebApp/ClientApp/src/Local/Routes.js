import {Switch, Route} from 'react-router';
import Home from "./Home";
import Compare from "./Comparison/Compare";
import Annotate from "./Annotation/Annotate";
import React from "react";
import NotFound from "./PageNotFound/NotFound";
import withCustomFetch from "./withCustomFetch";
import ImageGallery from "./Gallery/ImageGallery";

const MonacoEditor = React.lazy(() => import("@monaco-editor/react"));

export const Routes = ({fetch}) => {

    return (
        <Switch>
            <Route exact path="/">
                <Home/>
            </Route>
            <Route path="/compare">
                <Compare
                    MonacoEditor={MonacoEditor}
                    fetchFunction={fetch}
                />
            </Route>
            <Route path="/annotate">
                <Annotate
                    MonacoEditor={MonacoEditor}
                    fetchFunction={fetch}
                />
            </Route>
            <Route path="/gallery">
                <ImageGallery
                    fetchFunction={fetch}
                />
            </Route>
            <Route>
                <NotFound/>
            </Route>
        </Switch>
    );
};

export default withCustomFetch(fetch)(Routes);
