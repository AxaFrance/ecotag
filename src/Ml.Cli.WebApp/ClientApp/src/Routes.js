import {Switch, Route} from 'react-router';
import Home from "./Home";
import Compare from "./Comparison/Compare";
import Annotate from "./Annotation/Annotate";
import React from "react";
import NotFound from "./PageNotFound/NotFound";
import withCustomFetch from "./withCustomFetch";

export const Routes = ({fetch}) => {
    const MonacoEditor = React.lazy(() => import("@monaco-editor/react"));

    return (
        <Switch>
            <Route exact path="/">
                <Home/>
            </Route>
            <Route path="/compare">
                <Compare
                    MonacoEditor={MonacoEditor}
                    fetchFunction={fetch}
                />
            </Route>
            <Route path="/annotate">
                <Annotate
                    MonacoEditor={MonacoEditor}
                    fetchFunction={fetch}
                />
            </Route>
            <Route>
                <NotFound/>
            </Route>
        </Switch>
    );
};

export default withCustomFetch(fetch)(Routes);
