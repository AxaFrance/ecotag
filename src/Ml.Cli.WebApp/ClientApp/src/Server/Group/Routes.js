import React from 'react';
import { Switch, Route } from 'react-router-dom';
import PageNotFound from '../NotFound';

const Routes = () => {
  return (
    <Switch>
      <Route component={PageNotFound} />
    </Switch>
  );
};

export default Routes;
