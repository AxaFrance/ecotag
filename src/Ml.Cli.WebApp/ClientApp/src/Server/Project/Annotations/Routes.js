import React from 'react';
import { Switch, Route, useRouteMatch } from 'react-router-dom';
import AnnotationDispatchContainer from './AnnotationDispatch.container';

const Routes = () => {
  const { path } = useRouteMatch();
  return (
      
    <Switch>
      <Route path={`${path}/:documentId`} component={AnnotationDispatchContainer} />
    </Switch>
  );
};

export default Routes;
