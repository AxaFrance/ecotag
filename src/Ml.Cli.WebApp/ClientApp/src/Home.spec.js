import React from "react";
import '@testing-library/jest-dom';
import {fireEvent, render, waitFor} from '@testing-library/react';
import { Routes } from "./Routes";
import {BrowserRouter as Router} from "react-router-dom";


describe('Check page shifting', () => {
    test('Render home page and go to compare page', async () => {
        const { container, asFragment  } = render(<Router basename="/"><Routes /></Router>);

        expect(asFragment()).toMatchSnapshot();
        
        const compareButton = container.querySelector(".home__link");

        fireEvent.click(compareButton);

        await waitFor(() => expect(container.querySelector('.file-input')).not.toBeNull());
        expect(asFragment()).toMatchSnapshot();
    });
});
