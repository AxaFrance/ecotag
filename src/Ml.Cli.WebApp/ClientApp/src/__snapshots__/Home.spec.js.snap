// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check page shifting Render home page and go to compare page 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <div
      class="af-header"
    >
      <div
        class="container af-header__wrapper"
      >
        <header
          class="af-header__content"
          role="banner"
        >
          <div
            class="af-header__name"
          >
            <a
              class="af-logo"
              href="/"
            >
              <img
                alt="AXA Logo"
                class="af-logo__brand"
                src="logo-axa.svg"
              />
            </a>
            <h2
              class="af-header__title"
            >
              ML-CLI
              <span
                class="af-header__subtitle"
              >
                Made by AXA
              </span>
            </h2>
          </div>
        </header>
      </div>
    </div>
    <div
      class="af-title-bar"
    >
      <div
        class="container af-title-bar__wrapper"
      >
        <h1
          class="af-title-bar__title"
        >
          Services
        </h1>
      </div>
    </div>
    <div
      class="home__links-container"
    >
      <a
        class="home__link"
        href="/compare"
      >
        <div
          class="home__link-container home__link-container--compare"
        >
          Comparison
        </div>
      </a>
      <a
        class="home__link"
        href="/annotate"
      >
        <div
          class="home__link-container home__link-container--annotate"
        >
          Annotation
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Check page shifting Render home page and go to compare page 2`] = `
<DocumentFragment>
  <div
    class="af-header"
  >
    <div
      class="container af-header__wrapper"
    >
      <header
        class="af-header__content"
        role="banner"
      >
        <div
          class="af-header__name"
        >
          <a
            class="af-logo"
            href="/"
          >
            <img
              alt="AXA Logo"
              class="af-logo__brand"
              src="logo-axa.svg"
            />
          </a>
          <h2
            class="af-header__title"
          >
            ML-CLI
            <span
              class="af-header__subtitle"
            >
              Made by AXA
            </span>
          </h2>
        </div>
      </header>
    </div>
  </div>
  <div
    class="af-title-bar af-title-bar--backhome"
  >
    <div
      class="container-fluid container af-title-bar__wrapper"
    >
      <a
        class="btn af-btn--circle"
        href="/"
        title="Retour Ã  l'accueil"
      >
        <i
          class="glyphicon glyphicon-home"
        />
      </a>
      <h1
        class="af-title-bar__title"
      >
        Compare JSON file
      </h1>
    </div>
  </div>
  <div
    class="file-loader__container"
  >
    <div
      class="file-loader__input"
    >
      <div
        class="af-form__file-input"
      >
        <div
          class="drop-box hidden-mobile"
          id="file_loader"
          tabindex="0"
        >
          <input
            accept="application/json"
            autocomplete="off"
            multiple=""
            name="placeJsonFile"
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <div>
            Slide/Deposit files
          </div>
        </div>
        <button
          class="af-btn af-btn--file af-btn--hasiconLeft"
          type="button"
        >
          <i
            class="glyphicon glyphicon-open"
          />
           Browse
        </button>
      </div>
    </div>
    <div
      class="library__container"
    >
      <p
        class="library__title"
      >
        Files
      </p>
      <span>
        No file found.
      </span>
      <div
        class="library__files-list"
      />
    </div>
  </div>
  <div
    class="tabs"
  >
    <div
      class="tabs__header"
    >
      <ul
        class="af-tabs__control"
      >
        <li
          class="af-tabs__item af-tabs__item--active"
        >
          <button
            class="af-tabs__link"
            id="0"
          >
            Filters
          </button>
        </li>
        <li
          class="af-tabs__item"
        >
          <button
            class="af-tabs__link"
            id="1"
          >
            Time
          </button>
        </li>
        <li
          class="af-tabs__item"
        >
          <button
            class="af-tabs__link"
            id="2"
          >
            Scripts
          </button>
        </li>
        <li
          class="af-tabs__item"
        >
          <button
            class="af-tabs__link"
            id="3"
          >
            Configuration
          </button>
        </li>
      </ul>
      <div
        class="af-tabs__content"
      >
        <div
          class="af-tabs__pane"
        >
          <div
            class="tabs__container tabs__container--justify"
          >
            <div
              class="tabs__filter-container"
            >
              <p
                class="tabs__title"
              >
                File state:
              </p>
              <div
                class="af-form__select-container"
              >
                <select
                  class="af-form__input-select"
                  id="select_type"
                  name="SelectType"
                >
                  <option
                    value="KO"
                  >
                    KO
                  </option>
                  <option
                    value="OK"
                  >
                    OK
                  </option>
                  <option
                    value="All"
                  >
                    All
                  </option>
                </select>
                <span
                  aria-controls="select_type"
                  class="glyphicon glyphicon-menu-down"
                />
              </div>
              <p
                class="tabs__title"
              >
                Files extensions:
              </p>
              <div
                class="af-form__select-container"
              >
                <select
                  class="af-form__input-select"
                  id="extension_type"
                  name="ExtensionType"
                >
                  <option
                    value="All"
                  >
                    All
                  </option>
                  <option
                    value="PNG"
                  >
                    PNG
                  </option>
                  <option
                    value="JPG/JPEG"
                  >
                    JPG/JPEG
                  </option>
                  <option
                    value="PDF"
                  >
                    PDF
                  </option>
                  <option
                    value="TIFF"
                  >
                    TIFF
                  </option>
                </select>
                <span
                  aria-controls="extension_type"
                  class="glyphicon glyphicon-menu-down"
                />
              </div>
              <p
                class="tabs__title"
              >
                Server status:
              </p>
              <div
                class="af-form__select-container"
              >
                <select
                  class="af-form__input-select"
                  id="status_code_type"
                  name="StatusCodeType"
                >
                  <option
                    value="All"
                  >
                    All
                  </option>
                </select>
                <span
                  aria-controls="status_code_type"
                  class="glyphicon glyphicon-menu-down"
                />
              </div>
            </div>
            <div
              class="tabs__filter-container"
            >
              <p
                class="tabs__title"
              >
                Search Bar:
              </p>
              <textarea
                class="af-form__input-textarea"
                id="text_area_searchbar"
                name="TextAreaSearchBar"
                placeholder=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="file-input"
  >
    <button
      class="btn af-btn"
    >
      Apply filters
    </button>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
